### PR TITLE
Use logger names automatically derived from the containing class names

### PR DESCRIPTION
**Pull request by proski**
```
*  Use logger names automatically derived from the containing class names
   
   StashBuildListener and StashPullRequestsBuilder were using
   "StashBuildTrigger" instead of their own names.
   
   Co-developed-by: Anders Hammar <anders@hammar.net>
   Co-developed-by: jakub-bochenski <kuba.bochenski@gmail.com>
```
This is a superset of #45 based on top of the current master branch. All logger names are generated automatically using exactly the same code.

### CODE REVIEW

**Comment by proski** [comment]
The change made FindBugs detect that `StashBuilds#trigger` is unused (no idea why FindBugs could not see that issue before). I was aware of that and planned to address the issue in a separate commit. But since it's blocking the check, I added that fix here. I made it the first commit to avoid having an intermediate revision that doesn't pass FindBugs.

I have successfully tested both the PR merging functionality and the log names.

**Review by jimklimov — APPROVED** [review]

File: src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuilds.java [hunk 1]
```diff
@@ -94,8 +96,7 @@ public void onCompleted(AbstractBuild<?, ?> build, TaskListener listener) {
         duration);
 
     // Merge PR
-    StashBuildTrigger trig = StashBuildTrigger.getTrigger(build.getProject());
-    if (trig.getMergeOnSuccess() && build.getResult() == Result.SUCCESS) {
+    if (trigger.getMergeOnSuccess() && build.getResult() == Result.SUCCESS) {
```
Comment (jakub-bochenski) [inline src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuilds.java:99]: I didn't expect this based on the PR description :)
Are you 100% sure this is always the same?

**Comment by jakub-bochenski** [comment]
I guess this supersedes #45 ?

File: src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuilds.java (hunk 1 above)
Comment (jakub-bochenski) [reply src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuilds.java:99]: OK, just saw your comment above

**Review by jakub-bochenski — APPROVED** [review]

**Comment by jakub-bochenski** [comment]
> I guess this supersedes #45 ?

I'm bad at reading comments today

File: src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuilds.java (hunk 1 above)
Comment (proski) [reply src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuilds.java:99]: I have actually tested the merge functionality. The `trigger` variable comes from the constructor that is created by an instance of `StashBuildTrigger`. I believe it should be trusted.

**Pull request merged**